contracts/vcnote/VCNoteRouter.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.12;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {BorrowPermitParams} from "./libraries/BorrowPermitParams.sol";
+import {CErc20, CTokenInterface} from "../CErc20.sol";
+import {VCNote} from "./VCNote.sol";
+import {Upgradeable} from "../_proxy/Upgradeable.sol";
+import {VCNoteRouterStorageLib, VCNoteRouterStorage} from "./storages/VCNoteRouterStorage.sol";
+
+contract VivaRouter is Upgradeable {
+
+    function initialize(address _note, address _cNOTE, address _vcNOTE) external initializer {
+        VCNoteRouterStorageLib.set(_note, _cNOTE, _vcNOTE);
+
+        IERC20(_note).approve(_cNOTE, type(uint256).max);
+        IERC20(_cNOTE).approve(_vcNOTE, type(uint256).max);
+    }
+
+    function mint(uint256 amount) external {
+        VCNoteRouterStorage memory s = VCNoteRouterStorageLib.get();
+
+        uint256 balanceCNoteBefore = IERC20(s.cNOTE).balanceOf(address(this));
+        uint256 balanceVCNoteBefore = IERC20(s.vcNOTE).balanceOf(address(this));
+
+        // tansfer NOTE from the user to router contract
+        IERC20(s.NOTE).transferFrom(msg.sender, address(this), amount);
+        // mint cNOTE in clm
+        CErc20(s.cNOTE).mint(amount);
+
+        // mint vcNOTE in vivacity
+        uint256 mintedCNote = IERC20(s.cNOTE).balanceOf(address(this)) - balanceCNoteBefore;
+        CErc20(s.vcNOTE).mint(mintedCNote);
+
+        // transfer vcNOTE to the user
+        uint256 mintedVCNote = IERC20(s.vcNOTE).balanceOf(address(this)) - balanceVCNoteBefore;
+        IERC20(s.vcNOTE).transfer(msg.sender, mintedVCNote);
+    }
```

Comment:
What about emitting an event for this mint (also as redeem/repay as well below)?